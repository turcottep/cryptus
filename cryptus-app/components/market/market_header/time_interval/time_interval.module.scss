@use "/styles/variables" as v;
@import "/styles/utils";

.icon {
  position: relative;

  border-radius: 50%;
  width: v.$font-size-xl;
  height: v.$font-size-xl;
  font-size: v.$font-size-s;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  cursor: pointer;
}

.notactive {
  @extend .icon;
  background-color: white;
  color: #93918e;
}

.active {
  @extend .icon;
  @extend .cooltext;
  scale: 1.2;
}

.icon:hover:not(.active) {
  opacity: 1;
  visibility: visible;
  @extend .cooltext;
  scale: 1.1;
  cursor: pointer;
  transition: all 0.1s linear;
}

.container {
  flex: 0 0 none;

  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  gap: clamp(4px, 1vw, 16px);
}
