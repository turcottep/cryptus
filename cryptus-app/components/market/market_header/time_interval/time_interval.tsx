//react and css
import React, { useState, useEffect } from "react";
import s from "./time_interval.module.scss";

import { intervals } from "../../../../lib/data_types";

//external exports

//internal imports

//intervals enum

export default function TimeInterval(props: { active: number; callback: any }) {
  const [active, setActive] = useState(props.active);
  const select_interval = (index: number) => {
    setActive(index);
    props.callback(index);
  };

  const interval = (index: number) => {
    switch (index) {
      case intervals.week:
        return (
          <div
            className={
              s.icon + " " + (active == intervals.week ? s.active : "")
            }
            onClick={() => {
              select_interval(intervals.week);
            }}
          >
            1W
          </div>
        );
      case intervals.month:
        return (
          <div
            className={
              s.icon + " " + (active == intervals.month ? s.active : "")
            }
            onClick={() => {
              select_interval(intervals.month);
            }}
          >
            1M
          </div>
        );
      case intervals.three_months:
        return (
          <div
            className={
              s.icon + " " + (active == intervals.three_months ? s.active : "")
            }
            onClick={() => {
              select_interval(intervals.three_months);
            }}
          >
            3M
          </div>
        );
      case intervals.year:
        return (
          <div
            className={
              s.icon + " " + (active == intervals.year ? s.active : "")
            }
            onClick={() => {
              select_interval(intervals.year);
            }}
          >
            1Y
          </div>
        );
      case intervals.alltime:
        return (
          <div
            className={
              s.icon + " " + (active == intervals.alltime ? s.active : "")
            }
            onClick={() => {
              select_interval(intervals.alltime);
            }}
          >
            ALL
          </div>
        );
    }
  };

  return (
    <div className={s.container}>
      {interval(0)}
      {interval(1)}
      {interval(2)}
      {interval(3)}
      {interval(4)}
    </div>
  );
}
