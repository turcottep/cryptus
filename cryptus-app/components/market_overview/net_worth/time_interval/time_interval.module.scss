@use "/styles/variables" as v;

.icon {
  position: relative;
  background-color: #ffffff;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  font-size: 18px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  cursor: pointer;
}

.icon:hover:not(.active) {
  background-color: #ebe7e3;
  color: v.$text-color;
  opacity: 1;
  visibility: visible;
}

.active {
  background-color: #ebe7e3;
  color: #93918e;
  opacity: 1;
  visibility: visible;
}

.container_row {
  flex: 1 1 auto;
  display: flex;
  width: 100%;
  margin: 4px 0px;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  background-color: v.$bg-color;
}
