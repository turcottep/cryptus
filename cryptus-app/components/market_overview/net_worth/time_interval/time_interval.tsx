//react and css
import React, { useState, useEffect } from "react";
import s from "./time_interval.module.scss";

//external exports

//internal imports

export default function TimeInterval(props: { active: string }) {
  return (
    <div className={s.container_row}>
      <div className={s.icon}>1W</div>
      <div className={s.icon}>1M</div>
      <div className={s.icon}>3M</div>
      <div className={s.icon}>1Y</div>
    </div>
  );
}
