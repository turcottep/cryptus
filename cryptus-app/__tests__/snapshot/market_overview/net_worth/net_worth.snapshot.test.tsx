import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";

import { intervals } from "../../../../components/market/net_worth/time_interval/time_interval";
import NetWorth from "../../../../components/market/net_worth/net_worth";

describe("<NetWorth />", () => {
  it("should display properly", async () => {
    const test_props = {
      EthCad: 1,
      active: intervals.week,
      value: 7.79,
      delta: "0.01",
      callbackGraph: async () => {},
    };
    React.useState = jest
      .fn()
      .mockReturnValueOnce([test_props.active, {}])
      .mockReturnValueOnce([test_props.EthCad, {}]);

    const tree = renderer.create(<NetWorth {...test_props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
