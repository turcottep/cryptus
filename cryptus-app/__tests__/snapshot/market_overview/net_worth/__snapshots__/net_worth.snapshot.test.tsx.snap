// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NetWorth /> should display properly 1`] = `
<div>
  <div>
    networth: 7.79 eth
  </div>
  <div />
  <div>
    <div
      className="undefined undefined"
      onClick={[Function]}
    >
      1W
    </div>
    <div
      className="undefined "
      onClick={[Function]}
    >
      1M
    </div>
    <div
      className="undefined "
      onClick={[Function]}
    >
      3M
    </div>
    <div
      className="undefined "
      onClick={[Function]}
    >
      1Y
    </div>
    <div
      className="undefined "
      onClick={[Function]}
    >
      ALL
    </div>
  </div>
</div>
`;
